packages/loopring_v3/contracts/impl/libexchange/ExchangeDeposits.sol

```diff
@@ -99,7 +101,8 @@ library ExchangeDeposits
 
         // Transfer the tokens to this contract
         transferDeposit(
-            msg.sender,
+            S,
```

Comment:
I moved it as it can't hurt, but this is why all `external` state changing functions are guarded with `nonReentrant` so we don't have to worry as these things can get quite complicated in the exchange contract. And the tests automatically find all `external` state changing functions using the ABI so we don't have to worry about forgetting to add `nonReentrant` if we run the tests.